src/Microsoft.DotNet.XHarness.InstrumentationBase.Xunit/AndroidInstrumentationBase.cs

```diff
@@ -0,0 +1,192 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Reflection;
+using Android.App;
+using Android.OS;
+using Android.Runtime;
+using Microsoft.DotNet.XHarness.Common.CLI;
+using Microsoft.DotNet.XHarness.TestRunners.Common;
+using Microsoft.DotNet.XHarness.TestRunners.Xunit;
+
+// parsing of arguments for which test methods to run
+// properly set the return-code
+// to do the filter and excludes
+// automatically return the results file
+
+namespace Microsoft.DotNet.XHarness.InstrumentationBase.Xunit
+{
+    public abstract class AndroidInstrumentationBase : Instrumentation
+    {
+        private const string ResultsFileArgumentName = "results-file-name";
+        private const string ResultsPathArgumentName = "test-results-path";
+        private const string SummaryMessageArgumentName = "test-execution-summary";
+        private const string ExitCodeArgumentName = "return-code";
+        private const string ExcludeCategoriesDirArgumentName = "exclude-categories-dir";
+        private const string ExcludeCategoriesFileArgumentName = "exclude-categories-file";
+        private const string ExcludeMethodArgumentName = "exclude-method";
+        private const string IncludeMethodArgumentName = "include-method";
+        private const string ExcludeClassArgumentName = "exclude-class";
+        private const string IncludeClassArgumentName = "include-class";
+
+        private Bundle? _bundle;
+
+        protected Dictionary<string, string> bundleArguments = new Dictionary<string, string>();
+
+        protected AndroidInstrumentationBase(IntPtr handle, JniHandleOwnership transfer)
+            : base(handle, transfer)
+        {
+        }
+
+        protected virtual IEnumerable<Assembly> Tests
+        {
+            get
+            {
+                yield return Assembly.GetExecutingAssembly();
+            }
```

Comment:
I'm confuse by this property. Just returns the currently executing assembly. What is the goal? I'm not an android expert. But either:

1. We have on instrumentation with several test assemblies (each one of them with several tests).
2. We have an instrumentation class per assembly, with a number of tests.

If the second is correct, we could  modify the property to be just Assembly, else.. I don't understand why the class returns a IEnumerable.

Reply:
This property was aimed to pass tests from user to `TestsEntryPoint::GetTestAssemblies`. Anyway I'm going to remove `Instrumentation` class from this pull request to adjust it with #480